groundwork/patterns/gw_recipes_pattern.py

```diff
@@ -113,15 +116,23 @@ def __init__(self, app):
         self.__log = logging.getLogger(__name__)
         self.__log.info("Application recipes initialised")
 
-    def register(self, name, path, plugin, description=None, final_words=None):
+    def register(self, name, path, plugin, description=None, final_words=None,
+                 pre_hook=None, post_hook=None):
         """
         Registers a new recipe.
         """
         if name in self.recipes.keys():
             raise RecipeExistsException("Recipe %s was already registered by %s" %
                                         (name, self.recipes["name"].plugin.name))
 
-        self.recipes[name] = Recipe(name, path, plugin, description, final_words)
+        if not callable(pre_hook) and pre_hook is not None:
```

Comment:
I think you need to change the order.
First test if None, thenuse the variable for everything else.
Otherwise this could happen: if not callable(None)...

No sure, if callable() can handle None values